src/components/fhir/FhirProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import {oauth2 as SMART} from "fhirclient";
+import {FC, useState} from "react";
+import {FhirContext} from "./FhirContext";
+import Client from "fhirclient/lib/Client";
+
+/**
+ * Context provider to allow FHIR client to be passed through to child components.
+ * @param children children components to render
+ * @constructor
+ */
+const FhirProvider: FC<any> = ({children}) => {
+  const [client, setClient] = useState<Client | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const setClientOrError = (client: Client | null) => {
+    if (!client) {
+      SMART.ready()
+        .then((client) => setClient(client))
+        .catch((error) => setError(error))
+        .finally(() => console.debug("FHIR client ready"))
+    }
+  };
+
+  if (error) {
+    console.error(error.stack);
+    return <pre>{error.message}</pre>;
+  }
+
+  return (
+    <FhirContext.Provider value={{client: client, setClient: setClient}}>
+      <FhirContext.Consumer>
+        {({client}) => {
+          setClientOrError(client)
```

Comment:
Calling methods directly like `setClientOrError()` doesn't usually conform to the React way of doing things. We usually put such methods inside of a `useEffect` hook so that we can handle side effects elegantly. Again, we might be restricted by the library here, but I would put some thought into possible refactoring here.